pyannote/audio/applications/change_detection.py

```diff
@@ -0,0 +1,411 @@
+        step = config['sequences']['step']
+        balance = config['sequences']['balance']
+        generator = ChangeDetectionBatchGenerator(
+                feature_extraction,
+                duration=duration, step=step, batch_size=batch_size,balance=balance)
+
+        # number of samples per epoch + round it to closest batch
+        seconds_per_epoch = protocol.stats(subset)['annotated']
+        samples_per_epoch = batch_size * \
+                int(np.ceil((seconds_per_epoch / step) / batch_size))
+
+        # input shape (n_frames, n_features)
+        input_shape = generator.shape
+ 
+        labeling = SequenceLabeling()
+        labeling.fit(input_shape, architecture,
+                                 generator(getattr(protocol, subset)(), infinite=True),
+                                 samples_per_epoch, nb_epoch, loss='binary_crossentropy',
+                                 optimizer=optimizer, log_dir=train_dir)
+
+
+
+def evaluate(protocol, train_dir, store_dir, subset='development'):
+    os.makedirs(store_dir)
+
+    # -- LOAD MODEL --
+    architecture_yml = train_dir + '/architecture.yml'
+    WEIGHTS_H5 = train_dir + '/weights/{epoch:04d}.h5'
+    nb_epoch = 0
+    while True:
+        weights_h5 = WEIGHTS_H5.format(epoch=nb_epoch)
+        if not os.path.isfile(weights_h5):
+                break
+        nb_epoch += 1
+    config_dir = os.path.dirname(os.path.dirname(train_dir))
+    config_yml = config_dir + '/config.yml'
+    with open(config_yml, 'r') as fp:
+            config = yaml.load(fp)
+
+    # -- FEATURE EXTRACTION --
+    feature_extraction_name = config['feature_extraction']['name']
+    features = __import__('pyannote.audio.features',
+                                                fromlist=[feature_extraction_name])
+    FeatureExtraction = getattr(features, feature_extraction_name)
+    feature_extraction = FeatureExtraction(
+            **config['feature_extraction'].get('params', {}))
+
+    # -- SEQUENCE GENERATOR --
+    duration = config['sequences']['duration']
+    step = config['sequences']['step']
+
+    groundtruth = {}
+    for dev_file in getattr(protocol, subset)():
+                uri = dev_file['uri']
+                groundtruth[uri] = dev_file['annotation']
+
+    def objective_function(epoch):
+        weights_h5 = WEIGHTS_H5.format(epoch=epoch)
+        sequence_labeling = SequenceLabeling.from_disk(
+                architecture_yml, weights_h5)
+
+
+        aggregation = SequenceLabelingAggregation(
+                sequence_labeling, feature_extraction,
+                duration=duration, step=step)
+
+
+        predictions = {}
+
+        for dev_file in getattr(protocol, subset)():
+            uri = dev_file['uri']
+            #wav = dev_file['wav']
+            predictions[uri] = aggregation.apply(dev_file)
+
+        alphas = np.linspace(0, 1, 20)
+
+        purity = [SegmentationPurity(parallel=False) for alpha in alphas]
```

Comment:
Are you sure you want `parallel` set to False?
It could be faster to process every file in parallel.

But that's OK with me. No need to change it -- especially since I am not that confident with the quality of pyannote.metrics with respect to this parallel processing thing...